src/LevelAgreement.php

```diff
@@ -1164,4 +1164,39 @@ public function getLevelTicketClass(): string
     {
         return static::$levelticketclass;
     }
+
+    /**
+     * Remove level of previously assigned level agreements
+     *
+     * @return void
+     */
+    public function clearInvalidLevels(): void
+    {
+        // CLear levels of others LA of the same type
+        // e.g. if a new LA TTR was assigned, clear levels from others (= previous) LA TTR
+        $level_ticket_class = $this->getLevelTicketClass();
+        $level_class = $this->getLevelClass();
+        $levels = (new $level_ticket_class())->find([
```

Comment:
Is there a criteria missing to ensure only levels are cleared from a specific ticket?

Reply:
Indeed, good catch.
Looking into it.